### PR TITLE
Change string comparison into a .isEmpty() check

### PR DESCRIPTION
**Pull request by Juiceman**
We are actually checking if the string is empty. Using .isEmpty() is clearer.

### CODE REVIEW

File: src/freenet/client/filter/CSSTokenizerFilter.java [hunk 1]
```diff
@@ -1773,26 +1773,26 @@ else if(HTMLelement.indexOf('#')!=-1)
 			}
 
 		}
-		if(isIDSelector && "".equals(id)) return null; // No ID
+		if(isIDSelector && id.isEmpty()) return null; // No ID
```
Comment (ArneBab) [inline src/freenet/client/filter/CSSTokenizerFilter.java:1776]: are we sure here that id cannot be null? the check before was null-safe.

File: src/freenet/client/filter/CSSTokenizerFilter.java [hunk 2]
```diff
@@ -4502,9 +4502,9 @@ public boolean recursiveDoubleBarVerifier(String expression,ParsedWord[] words,F
 							}
 							// Against the rest of the pattern: the part that we've tried and failed plus the part that we haven't tried yet.
 							// NOT against the verifier we were just considering, because the double-bar operator expects no more than one match from each component of the pattern.
-							String pattern = ignoredParts+((("".equals(ignoredParts))||("".equals(secondPart)))?"":"a")+secondPart;
+							String pattern = ignoredParts+(((ignoredParts.isEmpty())||(secondPart.isEmpty()))?"":"a")+secondPart;
```
Comment (ArneBab) [inline src/freenet/client/filter/CSSTokenizerFilter.java:4505]: are we sure that ignoredParts and secondPart cannot be null?

**Comment by ArneBab** [comment]
Except for few cases where I’m not sure about null-safety, this looks good to me.

File: src/freenet/client/filter/CSSTokenizerFilter.java (hunk 1 above)
Comment (Bombe) [reply src/freenet/client/filter/CSSTokenizerFilter.java:1776]: Yes, this is either an empty string (assigned at declaration), or it’s the output of a `trim()` method call so this cannot be `null`.

File: src/freenet/client/filter/CSSTokenizerFilter.java (hunk 2 above)
Comment (Bombe) [reply src/freenet/client/filter/CSSTokenizerFilter.java:4505]: Yeah, same as above; assigned an empty string at declaration time, or assigned other values that are not `null`. :)

File: src/freenet/clients/fcp/DirPutFile.java [hunk 3]
```diff
@@ -50,7 +50,7 @@ public static DirPutFile create(SimpleFieldSet subset, String identifier, boolea
 		if(name == null)
 			throw new MessageInvalidException(ProtocolErrorMessage.MISSING_FIELD, "Missing field Name", identifier, global);
 		String contentTypeOverride = subset.get("Metadata.ContentType");
-		if(contentTypeOverride != null && (!contentTypeOverride.equals("")) && !DefaultMIMETypes.isPlausibleMIMEType(contentTypeOverride)) {
+		if(contentTypeOverride != null && (!contentTypeOverride.isEmpty()) && !DefaultMIMETypes.isPlausibleMIMEType(contentTypeOverride)) {
```
Comment (Bombe) [inline src/freenet/clients/fcp/DirPutFile.java:53]: The parentheses around the `isEmpty()` are unnecessary.

Comment (Juiceman) [reply src/freenet/clients/fcp/DirPutFile.java]: Fixed in [a29fa9d].
Thank you

**Comment by ArneBab** [comment]
Merged — thank you!

**Pull request merged**